### PR TITLE
🐛 fix TE with project not getting donated

### PR DESCRIPTION
**Pull request by khoi**
## What's this?
Fix a bug where TE with project associated not getting donated.

### Relationships

Introduced by #4359

## To reproduce

- Start a TE with a synced project attached
- The donation doesn't go through because of

```cs
if (!(timeEntry.ProjectId is long projectId))	
    return;
}
```

## Expected behavior

- The intent donation should work properly.

## Review hints
None.

## :squid: Permissions
🎙

### CODE REVIEW

**Comment by Odrakir** [comment]
I'm testing this... That check was there so entries for new projects are not donated, because we don't have the real (server) project ID yet, so we ignore those donations. Entries with project that were already synced before work as expected.

As far as I can tell the behaviour is correct and this doesn't need fixing. Am I missing something?

**Comment by khoi** [comment]
oops, the code was

```
if (!(timeEntry.ProjectId is long projectId)) {
   return
}
```

in cases where projectId is `null`, the function exits where it shouldn't be. 

This PR reverts the checking logic and add logic for building the `StartTimerIntent`'s `ProjectId`

**Comment by Odrakir** [comment]
I see, so this is a fix for entries without project. Understood.

File: Toggl.iOS/Services/IntentDonationServiceIos.cs [hunk 1]
```diff
@@ -58,20 +58,22 @@ public async Task DonateStartTimeEntry(IWorkspace workspace, ITimeEntry timeEntr
...
                 {
                     return;
                 }
 
-                if (projectId < 0 || timeEntry.TagIds.Any(tagId => tagId < 0))
+                if (timeEntry.ProjectId is long projectId)
                 {
-                    return;
+                    var project = await dataSource.Projects.GetById(projectId).FirstAsync();
+                    startTimerIntent.ProjectId = startTimerWithClipboardIntent.ProjectId = new INObject(projectId.ToString(), project.Name);
```
Comment (Odrakir) [inline Toggl.iOS/Services/IntentDonationServiceIos.cs]: I didn't realise about this before, but I'm not a big fan of multiple assignments in the same line. This is much harder to read than two lines.

File: Toggl.iOS/Services/IntentDonationServiceIos.cs [hunk 2]
```diff
@@ -58,20 +58,22 @@ public async Task DonateStartTimeEntry(IWorkspace workspace, ITimeEntry timeEntr
...
+                }
+                else
+                {
+                    startTimerIntent.ProjectId = startTimerWithClipboardIntent.ProjectId = new INObject(timeEntry.ProjectId.ToString(), String.Empty);
```
Comment (Odrakir) [inline Toggl.iOS/Services/IntentDonationServiceIos.cs]: Isn't `timeEntry.ProjectId` null here? Do we need this line at all?

**Pull request merged**